### PR TITLE
Upgrade to PHP 7.4

### PR DESCRIPTION
**Pull request by forevermatt**
### Changed
- Switch from `silintl/php7:7.2` to `php:7.4-apache` Docker base image

### Fixed
- Update docker-compose.yml to use a common, consistent base for PHP stuff
- Switch to maintained library for working with strings

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,31 +1,75 @@
...
+    docker-php-ext-enable gd pdo pdo_mysql mbstring xml curl
 
 # Make sure /data is available
 RUN mkdir -p /data
 
 # Copy in vhost configuration
 COPY build/vhost.conf /etc/apache2/sites-enabled/
 
+# If apache2 hasn't run there is no config file.
+CMD ["apache2ctl", "configtest"]
```
Comment (fillup) [inline Dockerfile:41]: I don't think you want this as `CMD` but maybe `RUN`? You have CMD again below which is the one you want running in the container.

File: Makefile [hunk 2]
```diff
@@ -14,7 +14,7 @@ bounce:
 
 clean:
 	docker-compose kill
-	docker-compose rm -f
+	docker system prune -f --volumes
```
Comment (fillup) [inline Makefile:17]: I prefer not to use `docker system prune` because that cleans up resources from other projects as well.

Comment (forevermatt) [reply Makefile:17]: Gotcha. Is it a problem if I leave it in this project? It's solved more problems than it's caused for me.

File: Dockerfile (hunk 1 above)
Comment (forevermatt) [reply Dockerfile:41]: Ohhh, thanks.

Comment (forevermatt) [reply Dockerfile]: Yep, that wasn't running. Found some issues to fix. 🙂

File: Makefile (hunk 2 above)
Comment (forevermatt) [reply Makefile:17]: Hmm... never mind. I'll check it back.

**Pull request merged**